objectstore/google.go

```diff
@@ -0,0 +1,302 @@
+package objectstore
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"errors"
+	"github.com/banzaicloud/banzai-types/components"
+	"github.com/banzaicloud/pipeline/auth"
+	"github.com/banzaicloud/pipeline/cluster"
+	"github.com/banzaicloud/pipeline/secret"
+	"github.com/gin-gonic/gin/json"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	apiStorage "google.golang.org/api/storage/v1"
+	"sort"
+	"strings"
+)
+
+// ManagedGoogleBucket is the schema for the DB
+type ManagedGoogleBucket struct {
+	ID           uint              `gorm:"primary_key"`
+	Organization auth.Organization `gorm:"foreignkey:OrgID"`
+	OrgID        uint              `gorm:"index;not null"`
+	Name         string            `gorm:"unique_index:bucketName"`
+	Location     string
+}
+
+// GoogleObjectStore stores all required parameters for container creation
+type GoogleObjectStore struct {
+	location       string
+	serviceAccount *cluster.ServiceAccount // TODO: serviceAccount type should be in a common place?
+	org            *auth.Organization
+}
+
+// NewGoogleServiceAccount creates a service account for google
+// TODO: this logic is duplicate thus should be in a common place so as it can be used from gke.go:newClientFromCredentials() as well
+func NewGoogleServiceAccount(s *secret.SecretsItemResponse) *cluster.ServiceAccount {
+	return &cluster.ServiceAccount{
```

Comment:
Maybe https://github.com/mitchellh/mapstructure could help here, it also adds validation into the mix.